velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -223,12 +256,28 @@ dwio::common::StripeProgress getStripeProgress(
  *
  * This method assumes each input `ColumnarBatch` have same schema.
  */
-void Writer::write(const VectorPtr& data) {
+void Writer::write(const VectorPtr& dataToWrite) {
+  VELOX_USER_CHECK(
+      dataToWrite->type()->equivalent(*schema_),
+      "The file schema type should be equal with the input rowvector type.");
+  auto rowVector =
+      std::dynamic_pointer_cast<facebook::velox::RowVector>(dataToWrite);
+  VELOX_CHECK_NULL(rowVector->nulls());
+  rowVector->setType(schema_);
```

Comment:
Remember that setType requires that 'schema_' and all its child vectors are singly-referenced. Can you guarantee this here?